sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -132,13 +132,12 @@ case class GetJsonObject(json: Expression, path: Expression)
     }
 
     if (parsed.isDefined) {
+      val parser = jsonFactory.createParser(jsonStr.getBytes)
+      val output = new ByteArrayOutputStream()
+      val generator = jsonFactory.createGenerator(output, JsonEncoding.UTF8)
```

Comment:
what if these things threw exceptions? Can you make them `var`s instead and do a null check in finally?